File: design-patterns/go/strategy/transport/transport_strategy.go
// transport_strategy.go

package main

type TransportStrategy struct {
	transport Transport
}

func NewTransportStrategy(transport Transport) (rcvr *TransportStrategy) {
	rcvr = &TransportStrategy{}
	rcvr.transport = transport
	return
}

func (rcvr *TransportStrategy) Execute() {
	rcvr.transport.Start()
	rcvr.transport.GetInfo()
	rcvr.transport.Operate()
}

func (rcvr *TransportStrategy) Repair() {
	rcvr.transport.Repair()
}

func (rcvr *TransportStrategy) Stop() {
	rcvr.transport.Stop()
}

File: design-patterns/php/strategy/transport/TransportStrategy.php
<?php
// TransportStrategy->php

namespace BigBoxCode\DesignPattern\Strategy\Transport;


class TransportStrategy {
    public function __construct(private Transport $transport) {
    }

    public function setStrategy(Transport $transport) {
        $this->transport = $transport;
    }
    
    public function execute(): void {
        $this->transport->start();

        $this->transport->getInfo();

        $this->transport->operate();
    }

    public function repair(): void {
        $this->transport->repair();
    }

    public function stop(): void {
        $this->transport->stop();
    }
}